python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -413,12 +414,32 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             config=self.config.photometry_ref_loader, log=self.log)
         self.photometry.match.setRefObjLoader(photometry_loader)
 
-        outputs = self.run(**inputs)
-
-        butlerQC.put(outputs, outputRefs)
+        # This should not happen with a properly configured execution context.
+        assert not inputs, "runQuantum got more inputs than expected"
+
+        result = pipeBase.Struct(exposure=None,
+                                 stars=None,
+                                 background=None,
+                                 psf_stars=None,
+                                 applied_photo_calib=None,
+                                 astrometry_matches=None,
+                                 photometry_matches=None,
+                                 )
```

Comment:
I'm torn as to whether it's better to spell out the possible fields here with `None` values, vs. letting `run` be the sole source of truth on those.  I think the `QuantumContext` changes in `pipe_base` should allow either to work.

Reply:
To simplify the call to `annotate` below (so it's not a mess of `getattr`), I need to at least specify the items that are passed to `annotate`. I don't like specifying a `result` list in both places (runQuantum and the top of run); I've changed it so `run` just makes an empty Struct (this caught a bug!) which is a bit better.

Reply:
This feels like it could be fragile if we ever change this task to output more things in the result Struct. Is it not possible to construct a list of arguments to feed the error annotation by iterating over all the existing things in the Struct?

Reply:
Not without using a whole bunch of `getattr` and tests of whether those things have a `metadata` property. In this design, the `PipelineTask` author has to specifically choose which output datasets could get annotated on failure and pass them to `annotate` individually. It might be nice to just pass the `result` struct and let `annotate` figure it out, but that's probably more complicated than it's worth (not all datasets support `metadata`, e.g. `AstropyArrow`).